assessments/projects/custodian/joint-review.md

```diff
@@ -0,0 +1,900 @@
+broader access into an environment and therefore fully replicate insider attack
+possibilities.
+
+### Example Attack Scenarios
+
+* Denial of Service, Tampering: Use lambda:PutFunctionConcurrency  to set a
+  Lambda Function's concurrency setting to 0 which would disable event based
+  compliance checking.
+* Information Disclosure: Redirecting Cloud Custodian log output
+  (logs:CreateLogStream, logs:PutLogEvents, logs:CreateLogGroup) to a
+  misconfigured or non-compliant S3 bucket (CLI: `custodian run --output-dir
+  s3://<my-bucket>/<my-prefix> <policyfile>.yml`), revealing to an attacker
+  detailed metadata about the cloud environment as well as positive Cloud
+  Custodian findings on how it is non-compliant (i.e. vulnerable).
+* Spoofing, Elevation of Privileges: Use lambda:CreateFunction and
+  lambda:InvokeFunction, thereby allowing for arbitrary code to be introduced
+  and executed when triggered.
+* Alternatively, use lambda:UpdateFunctionCode to update code for an existing
+  Lambda Function which would bypass the need for iam:PassRole. Then use
+  lambda:InvokeFunction to invoke whatever code that you want using the IAM role
+  that the Lambda Function is already attached to.  The iam:PassRole permissions
+  are only checked at function creation, not when the function is updated.
+* Attacker uses a Cloud Custodian Lambda function to leverage (iam:PassRole) to
+  change Ports on NACLs and Security Groups of PCI Cat 1 applications
+  (potentially leading to Information Disclosure).
+* Repudiation: User creates a Cloud Custodian Lambda function that assumes IAM
+  roles from Cloud Custodian to gain keys from the key management service.
+* Denial of Service: User creates two Cloud Custodian policies that run in
+  CloudTrail Mode (AWS lambdas are triggered by cloudtrail events to terminate
+  things). Example one policy that terminates instances with encryption turned
+  on and another that kills instances without encryption.
+
+### Security Degradation
+
+The project itself is a key target. With a small community, and even smaller
+approver and project review team, the likelihood that malicious policy content
+or code or dependencies will be released is non-zero.  This would significantly
+compromise users of Custodian who regularly update from the project git
+repository or use their python package distribution.
+
+The policy content maintained by the organization is a secondary target.
+Except at very large organizations, it is likely 1 or 2 users would maintain
+the entire policy repository with little to no oversight. It is unrealistic to
+expect smaller organizations to effectively review every policy change.
+
+The cloud permissions granted to Custodian are broad. Neither a secure default
+policy, nor policy generation or checker/linter tool, nor even adequate
+documentation exists to guide the new user on how to effectively lock down
+their policy to the absolute minimum.  Even then, the mixing of broad read and
```

Comment:
Recommend adding another Recommendation about the project that a secure default policy, linter/checker, and additional documentation be created to guide new users on locking down their policies.

Reply:
@TheFoxAtWork added to the CNCF ask specifically. please review